sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -106,22 +106,46 @@ private[sql] class RocksDBStateStoreProvider
     override def metrics: StateStoreMetrics = {
       val rocksDBMetrics = rocksDB.metrics
       def commitLatencyMs(typ: String): Long = rocksDBMetrics.lastCommitLatencyMs.getOrElse(typ, 0L)
-      def avgNativeOpsLatencyMs(typ: String): Long = {
-        rocksDBMetrics.nativeOpsLatencyMicros.get(typ).map(_.avg).getOrElse(0.0).toLong
+      def nativeOpsLatencyMillis(typ: String): Long = {
+        rocksDBMetrics.nativeOpsMetrics.get(typ).map(_ * 1000).getOrElse(0)
+      }
+      def avgNativeOpsLatencyNanos(typ: String): Long = {
+        (rocksDBMetrics.nativeOpsHistograms.get(typ).map(_.avg * 1000).getOrElse(0.0)).toLong
```

Comment:
Is it necessary to report at nano second-level?

Reply:
We have either millis or nanos for `SQLMetrics` [here](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala#L155). Given that this is avg, millis may not work.

Reply:
Thinking more about it, not sure if the avg makes sense anymore as we aggregate (sum) across the tasks and aggregaing (sum) of avg may be misleading. I think it should be the sum of the latency and couple this with the `get` count, we should be able to get the average across all tasks. Let me know what you think.

Reply:
Yea, aggregating sum of avg sounds not too much meaningful. An average overall tasks sounds more making sense.

Reply:
Changed it to use `sum` of latencies. And we already collect the number of `get` calls. Using these values at the driver side can give avg latencies.